bob-backend/src/pearl/disk_controller.rs

```diff
@@ -290,35 +290,34 @@ impl DiskController {
         Ok(groups)
     }
 
-    async fn find_group(&self, operation: &Operation) -> BackendResult<Group> {
-        Self::find_in_groups(self.groups.read().await.iter(), operation).ok_or_else(|| {
-            Error::failed(format!("cannot find actual alien folder. {:?}", operation))
-        })
+    async fn find_groups(&self, operation: &Operation) -> BackendResult<Vec<Group>> {
+        Ok(Self::find_in_groups(self.groups.read().await.iter(), operation))
     }
 
     fn find_in_groups<'pearl, 'op>(
-        mut pearls: impl Iterator<Item = &'pearl Group>,
+        pearls: impl Iterator<Item = &'pearl Group>,
         operation: &'op Operation,
-    ) -> Option<Group> {
+    ) -> Vec<Group> {
         pearls
-            .find(|group| group.can_process_operation(operation))
+            .filter(|group| group.can_process_operation(operation))
             .cloned()
+            .collect()
     }
 
-    async fn get_or_create_pearl(&self, operation: &Operation) -> BackendResult<Group> {
+    async fn get_or_create_first_pearl(&self, operation: &Operation) -> BackendResult<Group> {
         trace!("try get alien pearl, operation {:?}", operation);
         // block for read lock: it should be dropped before write lock is acquired (otherwise - deadlock)
         {
             let read_lock_groups = self.groups.read().await;
             let pearl = Self::find_in_groups(read_lock_groups.iter(), operation);
-            if let Some(g) = pearl {
+            if let Some(g) = pearl.first().cloned() {
```

Comment:
This is not good. Please, use old logic that returned single Group for this particular case